test/integration/count_test.lua

```diff
@@ -0,0 +1,679 @@
+        }, {
+            -- bucket_id is 1161, storage is s-2
+            id = 4, name = 'William', last_name = 'White',
+            age = 81, city = 'Chicago',
+        },
+    })
+
+    table.sort(customers, function(obj1, obj2) return obj1.id < obj2.id end)
+
+    local stat_a = storage_stat.collect(g.cluster)
+
+    -- Case: no conditions, just bucket id.
+    local result, err = g.cluster.main_server.net_box:call('crud.count', {
+        'customers',
+        nil,
+        {bucket_id = 2804, timeout = 1},
+    })
+    t.assert_equals(err, nil)
+    t.assert_equals(result, 1)
+
+    local stat_b = storage_stat.collect(g.cluster)
+    t.assert_equals(storage_stat.diff(stat_b, stat_a), {
+        ['s-1'] = {
+            requests = 1,
+        },
+        ['s-2'] = {
+            requests = 0,
+        },
+    })
+
+    -- Case: EQ on secondary index, which is not in the sharding
+    -- index (primary index in the case).
+    local result, err = g.cluster.main_server.net_box:call('crud.count', {
+        'customers',
+        {{'==', 'age', 81}},
+        {bucket_id = 1161, timeout = 1},
+    })
+    t.assert_equals(err, nil)
+    t.assert_equals(result, 1)
+
+    local stat_c = storage_stat.collect(g.cluster)
+    t.assert_equals(storage_stat.diff(stat_c, stat_b), {
+        ['s-1'] = {
+            requests = 0,
+        },
+        ['s-2'] = {
+            requests = 1,
+        },
+    })
+end
+
+pgroup.test_count_timeout = function(g)
+    -- let's insert five tuples on different replicasets
+    -- (two tuples on one replica and three on the other)
+    -- to check that  the total count will be calculated on the router
+    helpers.insert_objects(g, 'customers', {
+        {
+            -- bucket_id is 477, storage is s-2
+            id = 1, name = "Elizabeth", last_name = "Jackson",
+            age = 31, city = "New York",
+        }, {
+            -- bucket_id is 401, storage is s-2
+            id = 2, name = "Mary", last_name = "Brown",
+            age = 33, city = "Los Angeles",
+        },  {
+            -- bucket_id is 2804, storage is s-1
+            id = 3, name = "Elizabeth", last_name = "Smith",
+            age = 33, city = "Los Angeles",
+        }, {
+            -- bucket_id is 1161, storage is s-2
+            id = 4, name = "William", last_name = "White",
+            age = 81, city = "Chicago",
+        }, {
+            -- bucket_id is 1172, storage is s-1
+            id = 5, name = "Elizabeth", last_name = "May",
+            age = 28, city = "New York",
+        },
+    })
+
+    local conditions = {
+        {'<', 'age_index', 33},
+    }
+
+    local expected_len = 2
+    local timeout = 4
+    local begin = clock.proc()
+
+    local result, err = g.cluster.main_server.net_box:call('crud.count',
+            {'customers', conditions, {timeout = timeout}}
+    )
+
+    t.assert_equals(err, nil)
+    t.assert_equals(result, expected_len)
+    t.assert(clock.proc() - begin < timeout)
```

Comment:
perhaps we need for equality too.

Also luatest has a special asserts for 'greater that', 'less than' etc:

```


assert_lt (left, right[, message]) | Compare numbers.
-- | --
assert_le (left, right[, message]) |  
assert_gt (left, right[, message]) |  
assert_ge (left, right[, message])
```

Reply:
Updated:

```diff
pgroup.test_count_timeout = function(g)
    ...
-   t.assert(clock.proc() - begin < timeout)
+   t.assert_lt(clock.proc() - begin, timeout)
end
```